### PR TITLE
New: Validate Cloud Validators

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

Adds a cloud validator validator to make sure that cloud validators are formatted correctly.

Related issue: #7153

### Approach

Validates when a new trigger is created.

### TODOs before merging

- [x] Add test cases
- [x] Add entry to changelog

### CODE REVIEW

**Review by davimacedo — COMMENTED** [review]
Do you think it would be possible to also add at least one test case to make sure that nothing logs in when the configuration is valid?

**Comment by dblythy** [comment]
Good call, added in

**Review by davimacedo — APPROVED** [review]
LGTM!

**Comment by dblythy** [comment]
Needs to include new validator options `requireAllRoles` `requireAnyRoles`

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ ___
 - IMPROVE: Added new account lockout policy option `accountLockout.unlockOnPasswordReset` to automatically unlock account on password reset. [#7146](https://github.com/parse-community/parse-server/pull/7146). Thanks to [Manuel Trezza](https://github.com/mtrezza).
 - IMPROVE: Parse Server is from now on continuously tested against all recent MongoDB versions that have not reached their end-of-life support date. Added MongoDB compatibility table to Parse Server docs. [7161](https://github.com/parse-community/parse-server/pull/7161). Thanks to [Manuel Trezza](https://github.com/mtrezza).
 - IMPROVE: Parse Server is from now on continuously tested against all recent Node.js versions that have not reached their end-of-life support date. [7161](https://github.com/parse-community/parse-server/pull/7177). Thanks to [Manuel Trezza](https://github.com/mtrezza).
+- IMPROVE: Cloud Validators will now log on invalid configuration [#7154](https://github.com/parse-community/parse-server/pull/7154). Thanks to [dblythy](https://github.com/dblythy)
```
Comment (mtrezza) [inline CHANGELOG.md:29]: I have seen multiple names for this feature so far:
- cloud function validation
- Cloud Validators
- Parse.Cloud validator

Let's find a consistent name; it is validating `Cloud Functions` that are part of `Cloud Code`; in the docs this is already titled "cloud function validation", so let's stick with `Cloud Function validation`. How about:

> - IMPROVE: Log error on invalid Cloud Function validation configuration.

**Review by mtrezza — CHANGES_REQUESTED** [review]

**Comment by mtrezza** [comment]
Why is a failed validation only logging an error (that can be easily overseen) and not throwing an error on server start? I assume that in usual cases, validation configuration is not dynamically defined but statically defined once on server start.

**Comment by mtrezza** [comment]
Well, thanks for reacting so fast in changing to throwing, but I actually wanted to start a discussion about it to find a solution. What's your opinion on that?

**Comment by dblythy** [comment]
Sorry @mtrezza I’d thought that I’d replied to you. 

I didn’t realise that you meant a class to validate the Parse Server options as well. I think that’s a good idea and will make it a lot easier to add new features. 

I think throwing the error makes sense. The only concern is that it might break some configurations if they’ve accidentally mistyped a field.

**Comment by mtrezza** [comment]
I agree that a generic config validation doesn't have to be in this PR.

> The only concern is that it might break some configurations if they’ve accidentally mistyped a field.

Isn't mistyping one of the reasons why we'd want to throw?

**Comment by dblythy** [comment]
I mean that I would break existing configurations that have a mistyped key. It would effectively be a “breaking change”.

**Comment by mtrezza** [comment]
Introducing validation is not a breaking change as we do not assume misconfiguration as the basis reference to determine what constitutes a breaking change.

In other words, a misconfiguration is a developer bug, to which the same considerations apply as to a repository bug, which is also not assumed to be a basis reference to determine what constitutes a breaking change.

**Comment by mtrezza** [comment]
@dblythy Would you merge current master to get the tests to pass?

**Comment by mtrezza** [comment]
@dblythy Do you think this PR is ready, or anything you want to add?

**Review by mtrezza — APPROVED** [review]
LGTM

**Comment by dblythy** [comment]
> @dblythy Do you think this PR is ready, or anything you want to add?

I'm happy with it, thanks Manuel 😊

**Comment by mtrezza** [comment]
Great!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-server/releases/tag/5.0.0)

**Pull request merged**